packages/client/lib/AccessTokenClient/CredentialOfferAccessTokenClient.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  AccessTokenRequest,
+  AccessTokenResponse,
+  AuthorizationServerOpts,
+  CredentialOfferAccessTokenRequestOpts_V11,
+  EndpointMetadata,
+  IssuanceInitiationAccessTokenRequestOpts,
+  IssuerOpts,
+  OpenIDResponse,
+} from '@sphereon/openid4vci-common';
+
+import { IssuanceInitiationClient } from '../CredentialOffer';
+
+import { AccessTokenClient } from './AccessTokenClient';
+
+// noinspection JSUnusedLocalSymbols
+export class CredentialOfferAccessTokenClient implements AccessTokenClient {
```

Comment:
Why is this whole class needed according to you at this point? Where in the spec does V11 warrant a new implementation for the AccessTokenClient next to v9?

The more classes we create, the more maintenance and potential for bugs we introduce.

Also I rather see a suffix denoting the Version of the spec if we need seperate classes. The access token client itself has almost zero todo with Credential offers or Initiate Issuance. It is litterally an OAuth2 AccessToken client. By introducing these names it becomes confusing.

Reply:
All big things start small. And based on OO and design patterns this was supposed to be separate. This was also discussed on call with you and this approach was approved. Once the version 9 needs to be removed we will not need to surgically remove the version 9 specific details from single file.

I agree, it should have received only the data that the class needed rather than "initiation" object so that it should not have come to this.

Actually, I disagree with using union types ... As tomorrow for next version will we add another union type through out the code again ? 

@nklomp Shall I rename the files instead of removing these ?

Reply:
@nklomp Reverted.